sklearn/tree/_tree.pyx

```diff
@@ -168,32 +166,38 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
         cdef SIZE_t min_samples_split = self.min_samples_split
 
         # Recursive partition (without actual recursion)
-        splitter.init(X, y, sample_weight_ptr, X_idx_sorted)
+        cdef SIZE_t n_node_samples
+        cdef double weighted_n_node_samples
+        cdef double* sum_total
+        cdef double sq_sum_total
+        cdef double impurity
+        splitter.init(X, y, sample_weight_ptr, X_idx_sorted, 
+                      &n_node_samples, &weighted_n_node_samples, &sum_total,
+                      &sq_sum_total, &impurity)
 
         cdef SIZE_t start
         cdef SIZE_t end
         cdef SIZE_t depth
         cdef SIZE_t parent
         cdef bint is_left
-        cdef SIZE_t n_node_samples = splitter.n_samples
-        cdef double weighted_n_samples = splitter.weighted_n_samples
-        cdef double weighted_n_node_samples
         cdef SplitRecord split
         cdef SIZE_t node_id
 
         cdef double threshold
-        cdef double impurity = INFINITY
         cdef SIZE_t n_constant_features
         cdef bint is_leaf
         cdef bint first = 1
         cdef SIZE_t max_depth_seen = -1
         cdef int rc = 0
 
+        cdef double derp
```

Comment:
It was a helper variable I quickly used to make sure that my unit tests were passing because the code was correct, instead of because they hadn't properly compiled. I thought I had removed it.